packages/model-viewer/src/features/annotation.ts

```diff
@@ -132,9 +132,12 @@ export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
      * model-viewer element. The position and normal are returned as strings in
      * the format suitable for putting in a hotspot's data-position and
      * data-normal attributes. If the mesh is not hit, the result is null.
+     * 
+     * The third return value is the entire 3D object intersected at the pixel coordinates.
+     * This is usefull to apply material changes on mouse events (hover/click).
      */
     positionAndNormalFromPoint(pixelX: number, pixelY: number):
-        {position: Vector3D, normal: Vector3D}|null {
+      {position: Vector3D, normal: Vector3D, object?: any }|null {
```

Comment:
`any` is not going to work for us here. I'm pretty sure you'll find the type here is `THREE.Object3D`, but we want our public API to be agnostic of Three.js. An alternative would be to return our wrapped `Material` object instead, from the scene-graph API. How does that sound? FYI @timmmeh.

Reply:
It looks like this PR will be blocked until we have more of the SceneGraph support working which is in progress, I currently have PR #2816 that is going to let us address individual Three.js mesh primitives as wrapped Model-Viewer primitives, once this is in place we can create conversion helper from the Three Mesh/Primitive to ModelViewer Primitive from which you can get the ModelViewer material and return it.